app/src/modules/SignUp/Form.tsx

```diff
@@ -2,204 +2,258 @@ import React from "react";
-                    />
-                </Grid>
-
-                <Grid item xs={12} sm={6}>
-                    <TextField
-                        id="thirdName"
-                        name="thirdName"
-                        margin="normal"
-                        variant="outlined"
-                        label="Фамилия"
-                        autoComplete="Фамилия"
-                        required
-                        fullWidth
-
-                        error={Boolean(formValues.errors.thirdName)}
-                        helperText={formValues.errors.thirdName ? formValues.errors.thirdName : null}
-                        value={formValues.values.thirdName}
+	formValues: ISignUpState;
+	setFormValues: React.Dispatch<React.SetStateAction<ISignUpState>>;
+	onChange: (
+		field: string,
+		value: string,
+		validateCallback?: (value: string) => string,
+		anotherField?: string
+	) => void;
+}
 
-                        onChange={e => onChange("thirdName", e.target.value, validateFullName)}
-                    />
-                </Grid>
+export default React.memo(function SignUpForm({
+	formValues,
+	setFormValues,
+	onChange,
+}: ISignUpForm) {
+	const checkPassword = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setFormValues({
+			values: {
+				...formValues.values,
+				passwordConfirm: e.target.value,
+			},
+			errors: {
+				...formValues.errors,
+				password:
+					e.target.value &&
+					e.target.value === formValues.values.password
+						? ""
+						: e.target.value !== formValues.values.password
+						? "Введенные пароли не совпадают"
+						: REQUIRED_FIELD,
+				passwordConfirm:
+					e.target.value &&
+					e.target.value === formValues.values.password
+						? ""
+						: e.target.value !== formValues.values.password
+						? "Введенные пароли не совпадают"
+						: REQUIRED_FIELD,
+			},
+		});
+	};
 
-                <Grid item xs={12}>
-                    <TextField
-                        id="email"
-                        name="email"
-                        margin="normal"
-                        type="email"
-                        variant="outlined"
-                        label="Электронная почта"
-                        autoComplete="Электронная почта"
-                        required
-                        fullWidth
+	const validateFullName = (value: string) => {
+		return value
+			? value.length < 3
+				? "Длина поля должна быть не менее 3 символов"
+				: ""
+			: REQUIRED_FIELD;
+	};
 
-                        error={Boolean(formValues.errors.email)}
-                        helperText={formValues.errors.email ? formValues.errors.email : null}
-                        value={formValues.values.email}
+	const validateEmail = (value: string) => {
+		return value
+			? value.match(
+					/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```

Comment:
Maybe we can create a separate function to check the correct email address in utils.ts?